### PR TITLE
PLAT-1202 - Interactive `serverless` create

### LINKED ISSUE #4506: ~/.aws/credentials created on npm install (even if we do not rely on AWS)
# This is a Bug Report

After introduction of  #3866 due to [`ensureFileSync`](https://github.com/serverless/serverless/blob/192f80ccf4a0cdb1f062a1986be570f605829e74/lib/plugins/aws/configCredentials/awsConfigCredentials.js#L57-L58) run  in plugin constructor, the `~/.aws/credentials` file creation attempt is pursued on every SLS command, and also when serverless is installed due to [`serverless.init()`](https://github.com/serverless/serverless/blob/192f80ccf4a0cdb1f062a1986be570f605829e74/scripts/postinstall.js#L14) invoked in `scripts/postinstall.js`

It also happens when serverless is not intended to be used with AWS, therefore I believe it's a bug.

* ***Serverless Framework Version you're using***: 1.24.1
* ***Operating System***: 10.12.6
* ***Stack Trace***:
* ***Provider Error messages***:

### PR DESCRIPTION
**Pull request by medikoo**
Interactive `serverless` create, that allows to interactively setup a new service, or configure not yet configured feature.

Initially it proposes interactive setup of:

#### New service (from a chosen template).

_Only if command is invoked not in a service directory_.

At this point just two choices are proposed (`AWS Node.js` and `AWS Python`). This needs to be further discussed (I think current templates list is too large to propose them all)

#### AWS credentials

_Only If command is invoked in context of AWS service  and AWS credentials are not found to be setup in environment_

--- 

As this work overlapped some other functionalities, this PR also includes:
- Agnostic `lib/utils/createFromTemplate.js` util for creation of a new service from a template. `lib/plugins/create/create.js` was refactored to rely on that.
- Agnostic `lib/plugins/aws/utils/credentials.js` util for handling AWS credentials. ` lib/plugins/aws/configCredentials/awsConfigCredentials.js` was refactored to rely that, which brought in following improvments:
  - New profiles can be added via `config credentials` option (so far this option offered either complete overwrite or setup of new profile when non were configured)
  - Doesn't create `~/.aws/credentials` at package install and in case of non AWS services. That fixes #4506 
- `bin/serverless.js` - Cleanup
- `lib/classes/Service.js` - Improvement to options handling

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: lib/plugins/aws/configCredentials/awsConfigCredentials.test.js [hunk 1]
```diff
@@ -2,32 +2,28 @@
 
 const expect = require('chai').expect;
 const sandbox = require('sinon');
-const constants = require('constants');
+const constants = require('fs').constants;
```
Comment (pmuens) [inline lib/plugins/aws/configCredentials/awsConfigCredentials.test.js]: Yak-shaving, but why not `const { constants } = require('fs')` or moving the following right after the `require` statements `const { constants } = fs`.

File: lib/plugins/create/create.js [hunk 2]
```diff
@@ -225,49 +226,40 @@ class Create {
...
-
-      throw new this.serverless.classes.Error(errorMessage);
-    }
+          renameService(newServiceName, this.serverless.config.servicePath);
+        }
 
-    // rename the service if the user has provided a path via options and is creating a service
-    if ((boilerplatePath || serviceName) && notPlugin) {
-      const newServiceName = serviceName || boilerplatePath.split(path.sep).pop();
+        userStats.track('service_created', {
```
Comment (pmuens) [inline lib/plugins/create/create.js]: Oprhaned promise, but I guess this is still something we do in such situations.

Comment (medikoo) [reply lib/plugins/create/create.js]: Technically it's left the way it was (change is due to refactor that relocated that part in file).

Yes, we do it like that now. Idea is not bad, but the way it's implemented it's prone to issues (especially that we shortcut process with `process.exit`, so there's no guarantee those requests will fulfill, but it's not a concern for this PR - and we discuss it already in other channel)

**Comment by medikoo** [comment]
@pmuens I've updated this PR with:

- Fix AWS setup link
- EnsureServerless style
- Improve some formatting.

I've kept projects list to _AWS Node.js_ and _AWS Python_, as I still miss necessary feedback on how to expand it further. If we're to publish this with v1.47.0 I think it needs to stay that way.

File: lib/plugins/interactiveCli/setupAws.js [hunk 3]
```diff
@@ -75,8 +75,9 @@ module.exports = {
     return confirm('Do you want to set them up now?').then(isConfirmed => {
       if (!isConfirmed) return null;
       process.stdout.write(
-        '\nGo here to learn how to create your AWS credentials:\n' +
-          'https://slss.io/aws-account and enter them here:\n\n'
+        `\nGo here to learn how to create your AWS credentials:\n${chalk.bold(
+          'https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html#cli-quick-configuration-creds'
```
Comment (pmuens) [inline lib/plugins/interactiveCli/setupAws.js]: We should keep the `slss.io` URL as we've used it other places and can change the redirection URL  behind it without touching the Framework code.

Comment (medikoo) [reply lib/plugins/interactiveCli/setupAws.js]: Hmm.. but `https://slss.io/aws-account` is dead, and I remember that during our meeting we agreed that it's just a placeholder and should be replaced with valid one.

Do we really have an URL shortener available at https://slss.io/ ? If so who can set it up it with a valid AWS url?

Comment (pmuens) [reply lib/plugins/interactiveCli/setupAws.js]: Hmm. Interesting. TBH I don't know. I just remember that we used some short URLs in the past to control where the final URL redirects to and to further track how often such links are clicked.

Comment (medikoo) [reply lib/plugins/interactiveCli/setupAws.js]: Following conversation on Slack I think we should leave it the way it is

Comment (medikoo) [reply lib/plugins/interactiveCli/setupAws.js]: I just saw @skierkowski message on Slack and update it with he provided

**Pull request merged**